### ISSUE TITLE
Weird behaviour inside 'Group'

### ISSUE THREAD

**Issue by diegobernardes** [pressly/chi#169]
I want to override the functionality of **NotFound** and **MethodNotAllowed** and most of the time it works, but when i tried to put the handlers inside a group to add another middleware like the NoCache, it stopped working.

Moving the handlers outside of group make then work again.

```go
r.Group(func(r chi.Router) {
    r.Use(middleware.NoCache)

    r.NotFound(func(w http.ResponseWriter, r *http.Request) {
        fmt.Println("inside notFound handler")
        w.WriteHeader(http.StatusNotFound)
    })

    r.MethodNotAllowed(func(w http.ResponseWriter, r *http.Request) {
        fmt.Println("inside methodNotAllowed handler")
        w.WriteHeader(http.StatusMethodNotAllowed)
    })
})
```

**Comment by pkieltyka** [comment]
@diegobernardes @gpopovic I have a solution for this and I'll publish it soon

**Comment by pkieltyka** [comment]
fixed by 5b7406a71dbb35d4f59b0b48e68cd9f9256f5262

**Comment by diegobernardes** [comment]
wow that was fast! thanks @pkieltyka!
